kuma/javascript/src/index.jsx

```diff
@@ -25,49 +23,20 @@ if (container) {
     // Store the string catalog so that l10n.gettext() can do translations
     localize(data.locale, data.stringCatalog, data.pluralFunction);
 
-    let app = null;
-    // This switch statement is duplicated in ssr.jsx. Anything changed
-    // here should also be changed there. TODO: refactor this!
-    switch (componentName) {
-        case 'SPA':
-            // Ideally, we want as much as possible of MDN to be part
-            // of the single page app so that we can get client-side
-            // navigation between pages. Currently the single page app
-            // handles document pages and search results
-            app = (
-                <SinglePageApp
-                    initialURL={data.url}
-                    initialData={data.documentData}
-                />
-            );
-            break;
-        case 'landing':
-            // This is the React UI for the MDN homepage.
-            // The homepage has a React-based header, but most of the
-            // content is still based on Jinja templates, so we can't
-            // currently make it part of the single page app and have
-            // to handle it as a special case here.
-            app = <LandingPage />;
-            break;
-        case 'signupflow':
-            // This is the React UI for the MDN sign-up flow.
-            // The signup flow has a React-based header, but most of the
-            // content is still based on Jinja templates, so we can't
-            // currently make it part of the single page app and have
-            // to handle it as a special case here.
-            app = <SignupFlow />;
-            break;
-        default:
-            throw new Error(
-                `Cannot render or hydrate unknown component: ${componentName}`
-            );
+    let app = <App componentName={componentName} data={data} />;
+
+    if (app === null) {
```

Comment:
`React.createElement` for which JSX is syntactical sugar always returns an object, describing the component to be rendered (and its props). What will be null is the renderer output. E.g.:
![Screenshot (2)](https://user-images.githubusercontent.com/4051932/76229852-24437700-6223-11ea-89bc-d31d6f42119d.png)

One option would be to let the component itself throw and then add an error boundary for the SSR component. It'd also be great if we had tests that made sure it's correctly catching and logging the error in that case then.

Reply:
@Gregoor thanks for spotting this, and the comments. 

I've just added a second commit c878df1ec so it throws, along with a test to check ssr.jsx writes the caught error to console.

With adding an error boundary to the SSR component, I notice in index.jsx - the throw happens outside of the AppErrorBoundary, as it did before with the switch statement, I guess this is intentional, or should it actually be inside?

Reply:
Hmm, I think my comment above is partially incorrect, it seems in index.jsx, the error is not thrown when the App component is assigned, e.g.

```js
let app = <App componentName={componentName} data={data} />;
```

or when it is re-assigned to within the AppErrorBoundary, e.g.

```js
let app = (
  <GAProvider>
    <AppErrorBoundary>
      <React.StrictMode>{app}</React.StrictMode>
    </AppErrorBoundary>
  </GAProvider>
);
```

but when app is passed to render, e.g.
 
```js
ReactDOM.render(app, container);
```

So, I guess this PR's moves the error from being thrown outside of the AppErrorBoundary, to inside.

I'm not sure I understand why you would want to throw the error outside of it, but this does seem to change that.

Reply:
One more thing, using error boundaries in ssr.jsx, doesn't seem to be currently supported with `renderToString`: https://github.com/facebook/react/issues/10442